### PR TITLE
Add missing settings in tenant ApiController

### LINKED ISSUE #13108: Missing Tenant description setting in the Tenant's ApiController
### Describe the bug

Currently, the tenant's description setting is missing in the ApiController [here](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs#L109).

### Expected behavior
We should be able to set a tenant's description via an API call, so if this property is included in the sent model, the tenant should get this description.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #13108

### CODE REVIEW

**Comment by hishamco** [comment]
@MZole seems there are few other than description ;)

**Comment by hishamco** [comment]
What about recipes & feature profiles?

**Comment by MZole** [comment]
> @MZole seems there are few other than description ;)

Yes, good catch, I only noticed the description.

> What about recipes & feature profiles?

I have tested them and they work as intended, they get set from the sent model.

**Comment by hishamco** [comment]
> I have tested them and they work as intended, they get set from the sent model.

Really!! I didn't see any `IRecipeHarvester` injected there

**Comment by MZole** [comment]
Yes, I have tested it in our [project](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions), we have a console application tester for the `ApiClient` which creates and sets up a tenant, and I also tested your solution for the `isNewTenant` with this   :smiley:.

**Comment by hishamco** [comment]
Seems I need to try this one ;)

File: src/OrchardCore.Modules/OrchardCore.Tenants/ViewModels/CreateApiViewModel.cs [hunk 1]
```diff
@@ -2,6 +2,6 @@ namespace OrchardCore.Tenants.ViewModels
 {
     public class CreateApiViewModel : TenantViewModel
     {
-
+        public string Category { get; set; }
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Tenants/ViewModels/CreateApiViewModel.cs:5]: Maybe better to put it in the base `TenantViewModel` if needed in both derived `CreateApiViewModel ` and `EditTenantViewModel`.

**Pull request merged**